NorthstarDLL/mods/modmanager.cpp

```diff
@@ -398,10 +398,44 @@ void ModManager::LoadMods()
 	std::filesystem::directory_iterator classicModsDir = fs::directory_iterator(GetModFolderPath());
 	std::filesystem::directory_iterator remoteModsDir = fs::directory_iterator(GetRemoteModFolderPath());
 
+	this->m_invalidMods.clear();
+
 	for (std::filesystem::directory_iterator modIterator : {classicModsDir, remoteModsDir})
 		for (fs::directory_entry dir : modIterator)
+		{
 			if (fs::exists(dir.path() / "mod.json"))
 				modDirs.push_back(dir.path());
+			else if (fs::is_directory(dir.path()))
+			{
+				std::string filename = dir.path().filename().generic_string().c_str();
+				// Don't display an error for hidden directories
+				if (filename.at(0) == '.')
+					continue;
+
+				for (fs::directory_entry subdir : fs::recursive_directory_iterator(dir.path()))
+				{
+					fs::path modPath = subdir.path() / "mod.json";
+					if (fs::exists(modPath))
+					{
+						spdlog::warn(
+							"mod.json file for directory {} is located at the wrong location ({}).",
+							dir.path().generic_string().c_str(),
+							subdir.path().generic_string().c_str());
+						// read mod json file
+						std::ifstream jsonStream(modPath);
+						std::stringstream jsonStringStream;
+
+						while (jsonStream.peek() != EOF)
+							jsonStringStream << (char)jsonStream.get();
```

Comment:
Is there not a better way to just read the entire file to a string?

Reply:
I stole this one from [NorthstarLauncher existing code](https://github.com/R2Northstar/NorthstarLauncher/blob/6857b93f623dffedffac23fdd3983c74621d2b64/NorthstarDLL/mods/modmanager.cpp#L419); if you know a better way to do this, please let me know :)